### ISSUE TITLE
Please advise how to create seamless images

### ISSUE THREAD

**Issue by christopher5106** [crp94/textile#3]
I'm tried the textile metric to create some images that maximizes this score.

Natural image: textile score 0.18
![textile1](https://github.com/user-attachments/assets/23095519-cab2-492a-bf59-7a0cb07e13a2)

I've tried on both sides, reducing and augmenting the score

With score 0.9:
![textile10](https://github.com/user-attachments/assets/839366b3-d141-47b1-b324-080fe72bd8d8)

With score 0.004:
![textile11](https://github.com/user-attachments/assets/5a1ed308-113f-4dc4-a96c-a97a934b2eea)

I have no idea which one is tiling:
![image](https://github.com/user-attachments/assets/ce489cd4-b8c0-4a95-a8b9-73a98a4397b8)

![image](https://github.com/user-attachments/assets/c8e5418f-1805-4651-b2b2-9334f8d9edf9)

**Comment by crp94** [comment]
Hi Christopher! What are you doing exactly? It somehow looks like your optimization is adding noise in a way that is confusing our model (as in an adversarial attack)

**Comment by christopher5106** [comment]
Thanks for your help. Yes I wanted to create images that tile seamless and so I added the gradient to the noise prediction by Unet. Where is my error? Looks like an adversarial attack

**Comment by crp94** [comment]
It's hard to tell. What kind of diffusion process are you using? How are you normalizing your data? How many inference steps? How are you weighting the gradient?

**Comment by christopher5106** [comment]
I'm using a simple SDXL text2image pipeline, with correctly applied textile metric on the image correctly decoded from the latents for sure. What I'm  not sure is where to apply the gradients, I tried [here](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L1220) with

`latents = self.scheduler.step(noise_pred - lmbd * latents_grad, t, latents, **extra_step_kwargs, return_dict=False)[0]
`

but it's probably not the right place.

**Comment by crp94** [comment]
Hi! 

Yes, I think that makes sense. However, we never tried this with stable diffusion, so it may not actually work with that family of generative models. You may want to explore different weights to the gradients, or increasing the number of inference steps.
